inject/maven-plugin/src/main/java/io/helidon/inject/maven/plugin/AbstractApplicationCreatorMojo.java

```diff
@@ -16,80 +16,80 @@
-import io.helidon.inject.tools.ApplicationCreatorConfigOptions;
-import io.helidon.inject.tools.ApplicationCreatorRequest;
-import io.helidon.inject.tools.ApplicationCreatorResponse;
-import io.helidon.inject.tools.CodeGenFiler;
-import io.helidon.inject.tools.CodeGenPaths;
-import io.helidon.inject.tools.CompilerOptions;
-import io.helidon.inject.tools.ModuleInfoDescriptor;
-import io.helidon.inject.tools.PermittedProviderType;
-import io.helidon.inject.tools.ToolsException;
-import io.helidon.inject.tools.spi.ApplicationCreator;
-
-import org.apache.maven.model.Build;
+import io.helidon.inject.InjectionServices;
+import io.helidon.inject.Lookup;
+import io.helidon.inject.ServiceProvider;
+import io.helidon.inject.ServiceProviderBindable;
+import io.helidon.inject.ServiceProviderProvider;
+import io.helidon.inject.Services;
+import io.helidon.inject.codegen.InjectionCodegenContext;
+import io.helidon.inject.service.ModuleComponent;
+
+import org.apache.maven.artifact.Artifact;
 import org.apache.maven.plugins.annotations.Parameter;
 import org.apache.maven.project.MavenProject;
 
-import static io.helidon.inject.api.CallingContextFactory.globalCallingContext;
-import static io.helidon.inject.runtime.InjectionExceptions.toErrorMessage;
-import static io.helidon.inject.tools.ModuleUtils.REAL_MODULE_INFO_JAVA_NAME;
-import static io.helidon.inject.tools.ModuleUtils.isUnnamedModuleName;
-import static io.helidon.inject.tools.ModuleUtils.toBasePath;
-import static io.helidon.inject.tools.ModuleUtils.toSuggestedModuleName;
-import static java.util.Optional.ofNullable;
-
 /**
- * Abstract base for the Injection {@code maven-plugin} responsible for creating {@code Application} and Test {@code Application}'s.
+ * Abstract base for the Injection {@code maven-plugin} responsible for creating {@code Application} and Test
+ * {@code Application}'s.
  *
- * @see Application
- * @see ApplicationCreatorConfigOptions
+ * @see io.helidon.inject.Application
  */
-@SuppressWarnings({"unused", "FieldCanBeLocal"})
-public abstract class AbstractApplicationCreatorMojo extends AbstractCreatorMojo {
-
+abstract class AbstractApplicationCreatorMojo extends AbstractCreatorMojo {
+    private static final String MODULE_COMPONENT_CLASS_FILE_NAME = InjectionCodegenContext.MODULE_NAME + ".class";
     /**
      * The approach for handling providers.
      * See {@code ApplicationCreatorConfigOptions#permittedProviderTypes()}.
      */
-    @Parameter(property = "inject.permitted.provider.types", readonly = true)
-    private String permittedProviderTypes;
+    @Parameter(property = "inject.permitted.provider.types", defaultValue = "ALL")
     private PermittedProviderType permittedProviderType;
 
+    /**
+     * The -source argument for the Java compiler.
+     * Note: using the same as maven-compiler for convenience and least astonishment.
+     */
+    @Parameter(property = "maven.compiler.source", defaultValue = "21")
```

Comment:
these 21 constants should be extracted javadoc'ed

Reply:
Maven plugin documentation should show the default values from code (also tried in IDE - I can see the default value correctly).

Reply:
Also this was copied from your code in `AbstractCreatorMojo` that was not documented either. Extracting a constant does not really improve documentation or readability in this case (as it is used only in this one place)